### PR TITLE
Compatibility with WP 5.3: upload failures

### PR DESCRIPTION
**Pull request by Screenfeed**
WP 5.3 introduces a process to recover from media upload failures. Basically, if php crashes (not enough memory, etc), an ajax request will be done to create the missing thumbnails.
Imagify will trigger the auto-optimization once all thumbnails are created.

### CODE REVIEW

File: inc/classes/class-imagify-auto-optimization.php [hunk 1]
```diff
@@ -94,8 +117,11 @@ public function remove_hooks() {
 			remove_filter( 'big_image_size_threshold',             [ $this, 'prevent_auto_optimization_when_generating_thumbnails' ], $prio );
 			remove_filter( 'wp_generate_attachment_metadata',      [ $this, 'allow_auto_optimization_when_generating_thumbnails' ], $prio );
 			remove_action( 'imagify_after_auto_optimization_init', [ $this, 'do_auto_optimization' ], $prio );
+			// Upload failure handling.
+			remove_action( 'wp_ajax_media_create_image_subsizes',  [ $this, 'prevent_auto_optimization_when_recovering_from_upload_failure' ], 5 );
```
Comment (Tabrisrp) [inline inc/classes/class-imagify-auto-optimization.php]: Priority here is `5`, while on the `add_action` it was `-5`

Comment (Screenfeed) [reply inc/classes/class-imagify-auto-optimization.php]: Nice catch.

File: inc/classes/class-imagify-auto-optimization.php [hunk 2]
```diff
@@ -426,11 +457,101 @@ public function prevent_auto_optimization_when_generating_thumbnails( $threshold
...
+		static::prevent_optimization_internally( $attachment_id );
+
+		// Auto-optimization will be done on shutdown.
+		ob_start( [ $this, 'maybe_do_auto_optimization_after_recovering_from_upload_failure' ] );
+	}
+
+	/**
+	 * Maybe launch auto-optimization after recovering from an upload failure, when all thumbnails are created.
+	 *
+	 * @since  1.9
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/classes/class-imagify-auto-optimization.php]: And you think right. Copy/paste drama.
Thanks :)

File: inc/classes/class-imagify-auto-optimization.php (hunk 1 above)
Comment (Screenfeed) [reply inc/classes/class-imagify-auto-optimization.php]: It has been fixed.

**Pull request merged**